draft-ietf-mls-protocol.md

```diff
@@ -5390,6 +5469,23 @@ be mitigated by having clients regularly generate new KeyPackages and upload
 them to the Delivery Service.  This way, the key material used to add a member
 to a new group is more likely to be fresh and less likely to be compromised.
 
+## Uniqueness of Ratchet Tree Key Pairs
+
+The encryption and signature keys stored in ratchet tree nodes MUST be distinct
+from one another.  If two members' leaf nodes have the same signature key, then
+the data origin authentication properties afforded by signatures within the
+group are degraded.  If two nodes have the same encryption key, it creates a
+"double join" scenario where a removed member may still be able to compute the
+group's secrets.
```

Comment:
I'm not sure I get this argument. First, keys could be related in different ways, e.g. `pk2 = pk1 ^ 2`. If I know `sk1` corresponding to `pk1`, I can decrypt ciphertexts for `sk2 = 2 * sk1` corresponding to `pk2` as well. In general, such relations can't be detected, so I don't see how this check _can_ improve security. On the other hand, it's a bit annoying to implement.

Another reason I don't think this check prevents double joins by malicious insiders. Say leaves A and B have the same HPKE key. This could be because:

* A is honest and a malicious B copied A's key. Since B doesn't know the secret key, this is not an attack and B can be removed.
* A is honest and generated the same key as a malicious B. This clearly can't happen because keys are random.
* A and B are malicious, they collude and they introduced the same key. In this case security is restored only after removing both insiders. (Since A and B collude, they are one entity in a way)